bin/crystal

```diff
@@ -94,27 +94,22 @@ _canonicalize_file_path() {
 
 ##############################################################################
 
-# Based on http://stackoverflow.com/a/2108540/2199687
+# Based on http://stackoverflow.com/q/370047/641451
 remove_path_item() {
-  local path item work
+  local path item
 
   path="$1"
   item="$2"
 
-  work=":$path:"
-  work="${work/:$item:/:}"
-  work="${work%:}"
-  work="${work#:}"
-
-  echo "$work"
+  echo -n $path | awk -v RS=: -v ORS=: '$0 != "'$item'"' | sed 's/:$//'
```

Comment:
I came to it through [this guide](https://wiki.ubuntu.com/DashAsBinSh#A.24.7Bparm.2F.3F.2Fpat.5B.2Fstr.5D.7D) and the answer I put in the comment.

It doesn't seem doable without `awk`/`sed`/`perl`/`tr` - and I think `awk` is the less rare of them. But we can totally change it.

I don't have much experience with this minimalist systems - I've just targeted the Docker build and made sure it worked.